src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -505,29 +515,41 @@ private static bool LineIsEmpty(ArraySegment<byte> line)
 
         private void ParseStatusLine(Span<byte> line, HttpResponseMessage response)
         {
+            // We sent the request version as either 1.0 or 1.1.
+            // We expect a response version of the form 1.X, where X is a single digit as per RFC.
+
             if (line.Length < 14 || // "HTTP/1.1 123\r\n" with optional phrase before the crlf
                 line[0] != 'H' ||
                 line[1] != 'T' ||
                 line[2] != 'T' ||
                 line[3] != 'P' ||
                 line[4] != '/' ||
+                line[5] != '1' ||
+                line[6] != '.' ||
                 line[8] != ' ')
             {
                 ThrowInvalidHttpResponse();
             }
 
-            // Set the response HttpVersion and status code
-            byte majorVersion = line[5], minorVersion = line[7];
-            byte status1 = line[9], status2 = line[10], status3 = line[11];
-            if (!IsDigit(majorVersion) || line[6] != (byte)'.' || !IsDigit(minorVersion) ||
-                !IsDigit(status1) || !IsDigit(status2) || !IsDigit(status3))
+            // Set the response HttpVersion
+            byte minorVersion = line[7];
+            if (!IsDigit(minorVersion))
```

Comment:
If we do my above suggestion, we can remove the digit check, as at that point we know it's already a 0 or 1.

If we don't do my suggestion, we can at least avoid doing the check until we know it's not a 0 or 1, and thus in 99.99% of the cases we'll not do the check, e.g.
```C#
byte minorVersion = line[7];
response.Version =
    minorVersion == '1' ? HttpVersion.Version11 ?
    minorVersion == '0' ? HttpVersion.Version10 ?
    !IsDigit(minorVersion) ? throw ... ?
    new Version(1, minorVersion - '0');
```